pkg/generators/openapi.go

```diff
@@ -81,19 +81,25 @@ func hasOpenAPITagValue(comments []string, value string) bool {
 	return false
 }
 
-func hasRequiredTag(m *types.Member) bool {
-	return types.ExtractCommentTags(
-		"+", m.CommentLines)[tagRequired] != nil
-}
-
-// hasOptionalTag returns true if the member has +optional in its comments or
-// omitempty in its json tags.
-func hasOptionalTag(m *types.Member) bool {
+// isOptional returns error if the member has +optional and +required in
+// its comments. If +optional is present it returns false. If +required is present
+// it returns true. Otherwise, it returns true if `omitempty` JSON tag is present
```

Comment:
```suggestion
// its comments. If +optional is present it returns true. If +required is present
// it returns false. Otherwise, it returns true if `omitempty` JSON tag is present
```

Reply:
gah, ty